src/goto-symex/symex_assign.cpp

```diff
@@ -350,6 +358,92 @@ void symex_assignt::assign_from_struct(
   }
 }
 
+void symex_assignt::assign_constants(
+  const ssa_exprt &length,
+  const constant_exprt &new_length,
+  const ssa_exprt &char_array,
+  const array_exprt &new_char_array)
+{
+  // assign length of string
+  assign_symbol(
+    length,
+    expr_skeletont{},
+    new_length,
+    {});
+
+  symbol_table_baset &st_mutable =
+    const_cast<symbol_table_baset &>(ns.get_symbol_table());
+
+  array_typet new_char_array_type(
+    char_type,
+    new_length);
+
+  symbolt &aux_symbol = get_fresh_aux_symbol(
+    new_char_array_type,
+    "",
+    "char_array_constant",
+    source_locationt(),
+    ID_java,
+    st_mutable);
+
+  CHECK_RETURN(aux_symbol.is_state_var);
+  CHECK_RETURN(aux_symbol.is_lvalue);
+
+  aux_symbol.is_static_lifetime = true;
+  aux_symbol.is_file_local = false;
+  aux_symbol.is_thread_local = false;
+
+  aux_symbol.value = new_char_array;
+
+  exprt arr =
+    state
+      .rename(
+        aux_symbol.symbol_expr(),
+        ns)
+      .get();
+
+  assign_symbol(
+    to_ssa_expr(arr).get_l1_object(),
+    expr_skeletont{},
+    new_char_array,
+    {});
+
+  address_of_exprt aoe(
+    index_exprt(
+      aux_symbol.symbol_expr(),
+      from_integer(0, length_type)));
+
+  assign_symbol(
+    char_array,
+    expr_skeletont{},
+    aoe,
+    {});
+}
+
+bool symex_assignt::constant_propagate_empty_string(
+  const function_application_exprt &f_l1)
+{
+  const constant_exprt length = from_integer(0, length_type);
+
+  array_typet char_array_type(
+    char_type,
+    length);
+
+  INVARIANT(
+    f_l1.arguments().size() == 2,
+    "empty string primitive does not take arguments");
```

Comment:
not very clear because it asserts there are 2 arguments, but the comment says there are 0